### ISSUE TITLE
New windows invisible

### ISSUE THREAD

**Issue by fuzebox1** [paperwm/PaperWM#306]
All new windows i create, except for the web browser, are invisible and can only be seen if moved to scratch then back.
Running Arch, latest gnome, the AUR package of paperwm.

**Comment by hedning** [comment]
Did it start happening after a system update? Arch is often a few commits ahead of the tagged released so that could be it (though I took a look at the commit they ship and didn't see anything obvious pop out).

What's your monitor setup? (how many, and if you have `workspaces-only-on-primary` turned on or not?)

I'm guessing windows are shown while navigating/animating (eg. `super-right/left`), and the tiling doesn't respond to resizing invisible windows?

Not really sure what can be causing the second monitor to not show windows though.

**Comment by fuzebox1** [comment]
workspaces-only-on-primary is on. they are show while moving with super left right.
When i take a window from scratch back to the tiling, scratched windows on second monitor minimize.
GNOME Shell 3.36.2. 

I've also noticed that paperwm (at least it appers to be) causes some severe lag in wine games. And the top gnome menu overlays the full screen game.

**Comment by hedning** [comment]
Right, `workspaces-only-on-primary` doesn't really work well in paperwm (eg. windows to pushed into a secondary monitor will get detached), I'd reccomend turning it off (that will give you a workspace per monitor). It's been a while since I tried testing with it on, but it wouldn't surprise if a bug interacting with it have snuck in. 

> I've also noticed that paperwm (at least it appers to be) causes some severe lag in wine games

I'd thought those things had mostly been ironed out by ba059a07d62410561aca897975c6d7704678eafa, but yeah, the way we handle animations has the potential of causing some overhead.

> And the top gnome menu overlays the full screen game.

I'd guess this is a wine specific bug, I'll see if I get time to debug it.

**Comment by hedning** [comment]
If setting `workspaces-only-on-primary` to false didn't work you might be seeing a X11 specific bug (assuming you're on X11). Seen some weirdness the last week with gnome-shell 3.34.

**Comment by jtaala** [comment]
Is this still happening for anyone?  I haven't seen this happen.  Am assuming due to an old bug or change in either your distro, gnome, or an older PaperWM implementation.

**Comment by petterssonjonas** [comment]
I've not used paperwm for years, so no idea.

**Comment by jtaala** [comment]
> I've not used paperwm for years, so no idea.

All good @fuzebox1, thanks for replying.  Cheers :+1:

**Comment by jtaala** [comment]
Closing this issue as no longer valid, given no confirmed cases after a month.
